frontend/js/src/settings/music-services/details/MusicServices.tsx

```diff
@@ -46,24 +52,62 @@ export default function MusicServices() {
     soundcloud: loaderData.current_soundcloud_permissions,
     appleMusic: loaderData.current_apple_permissions,
     lastfm: loaderData.current_lastfm_permissions,
+    funkwhale: loaderData.current_funkwhale_permission,
     librefm: loaderData.current_librefm_permissions,
   });
 
+  const [funkwhaleHostUrl, setFunkwhaleHostUrl] = React.useState("");
+  const [
+    connectedFunkwhaleServers,
+    setConnectedFunkwhaleServers,
+  ] = React.useState<string[]>(loaderData.funkwhale_host_urls || []);
+
+  // Track if this is the initial load to prevent automatic disable calls
+  const [isInitialLoad, setIsInitialLoad] = React.useState(true);
+  const [userInteractionStarted, setUserInteractionStarted] = React.useState(
+    false
+  );
+
   const handlePermissionChange = async (
     serviceName: string,
     newValue: string
   ) => {
     try {
-      const response = await fetch(
-        `/settings/music-services/${serviceName}/disconnect/`,
-        {
-          method: "POST",
-          body: JSON.stringify({ action: newValue }),
-          headers: {
-            "Content-Type": "application/json",
-          },
-        }
-      );
+      setUserInteractionStarted(true);
+
+      // For Funkwhale disable action during initial load, don't call backend
+      // This prevents automatic disconnection when component loads with "disable" state
```

Comment:
Despite the comments I don't understand why this extra logic and state value are needed. Can you talk me through it? there might be a cleaner way to implement this.

I might be misunderstanding the issue, but setUserInteractionStarted should not be called on page load, and for FW it seems to only be called for the disconnection action.